Detectors/CTP/Raw/src/CTPRawWriter.cxx

```diff
@@ -0,0 +1,49 @@
+#include "CTPRaw/CTPRawWriter.h"
+#include "DataFormatsCTP/CTPRawData.h"
+#include "DetectorsRaw/RDHUtils.h"
+#include "CommonUtils/StringUtils.h"
+
+using namespace o2::ctp;
+ClassImp(CTPRawWriter);
+/**
+ * @brief CTPRawWriter::init
+ * CTP has one CRU and 3 GBT links
+ * - Interaction Record
+ * - Trigger Class Record
+ * - other: counters, HB map, ...
+ * Code based on Raw/test/testRawReaderWriter.cxx
+ */
+void CTPRawWriter::init()
+{
+  mWriter.useRDHVersion(6);
+  std::string outFileName = o2::utils::concat_string("ctp_testdata_", std::to_string(mcru), ".raw");
+  mWriter.registerLink(mlink0, mcru, mlink0, 0, outFileName);
+  mWriter.registerLink(mlink1, mcru, mlink1, 0, outFileName);
+}
+
+int o2::ctp::CTPRawWriter::createRawFromIRs()
+{
+  // Generate interactions
+  std::vector<o2::InteractionTimeRecord> irs(1000);
+  o2::steer::InteractionSampler irSampler;
+  irSampler.setInteractionRate(15000); // ~1.5 interactions per orbit
+  irSampler.init();
+  irSampler.generateCollisionTimes(irs);
+  //
```

Comment:
is this just a test? Should not you use the same interactions sample as used for digitization?

Reply:
Yes, it is just the test. I wanted to create data with RawWriter and read it with my RawReader.
Where in O2 is digitization happens ?
CTP 'digits' would be created from trigger inputs to CTP at the and but before I may want something simpler ...

Reply:
The digitization happens in the ``o2-sim-digitizer-workflow``, see  https://github.com/AliceO2Group/AliceO2/blob/dev/doc/DetectorSimulation.md#documentation-of-the-digitization-step-
There every detector creates a vector of digits (and for you, the relevant detectors create vector of triggerInputs) which are sent to writers. CTP should add a device which instead of using the hits from generators, subscribes to these triggerInput messages and produces its own "digits" (trigger decision) and send them to its own writer.
As an exercise you can make your own digit producer which will write a vector of your digits, then you will need to add for CTP an executable which creates CRU-formatted raw data, like e.g. https://github.com/AliceO2Group/AliceO2/blob/dev/Detectors/FIT/FT0/simulation/src/digi2raw.cxx
But you don't need RawFileReader to read these raw data. 
They can be either replayed by readout.exe / DataDistribution (complex way, there are some instructions [here](https://alice.its.cern.ch/jira/browse/O2-1492)), or using ``https://github.com/AliceO2Group/AliceO2/tree/dev/Detectors/Raw#raw-data-file-reader-workflow``, which will parse TFs from your raw data and feed them to your raw->digits decoded.

Reply:
Thanks. Working in that direction.